### ISSUE TITLE
feat: SIGN neural network

### ISSUE THREAD

**Issue by andreostrovsky** [graphframes/graphframes#907]
Please consider implementing [SIGN: Scalable Inception Graph Neural Networks](https://grlplus.github.io/papers/77.pdf). See this Medium article by the paper's authors for a good explanation: [Article](https://medium.com/data-science/simple-scalable-graph-neural-networks-7eb04f366d07), [Article without Medium paywall](https://freedium-mirror.cfd/https://medium.com/data-science/simple-scalable-graph-neural-networks-7eb04f366d07).

I think this makes a good candidate for a GNN suitable for implementation in GraphFrames. Many of the motivations and benefits discussed in the above articles align closely with @SemyonSinchenko's [blog post](https://semyonsinchenko.github.io/ssinchenko/post/graphframes-embeddings/) about graph embeddings at scale.

The core part of the model is a pre-computed transformation that appears to be easily distributable. Here's the [PyG implementation of the transform](https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/transforms/sign.html), and a [full usage example](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/sign.py).

The MLP portion could potentially just reuse the Spark MLlib MultilayerPerceptronClassifier, even though the details are not quite the same as those used in the paper.

If implemented, this could provide GraphFrames users a stronger method for node classification than the current two: word2vec (transductive, not inductive) and hash2vec (not as effective).

**Component**

- [ ] Scala Core Internal
- [ ] Scala API
- [ ] Spark Connect Plugin
- [ ] Infrastructure
- [ ] PySpark Classic
- [ ] PySpark Connect

**Additional context**

**Are you planning on creating a PR?**

- [ ] I'm willing to make a pull-request

**Comment by SemyonSinchenko** [comment]
I can implement only the CPU-based version (no GPU). Triangle-based operator I better would not to add: counting triangles at scale is a complex problem. PPR looks feasible as well as normalized matrix itself. While it is possible to add an MLP from SparkML (or made our own using netlib.BLAS), I'm thinking about keeping only preprocessing in GraphFrames while the final MLP training can be done in PyTorch or whatever. Maybe two options: end2end or just compute matrices and do a features preprocessing.
